src/lib.rs

```diff
@@ -163,53 +172,175 @@ pub mod pallet {
+			let data = call.encode();
+			let deposit = <BalanceOf<T>>::from(data.len() as u32)
+				.saturating_mul(T::PreimageByteDeposit::get());
+
+            T::Currency::reserve(&who, deposit)?;
+
+			let preimage = PreimageCall::<T::AccountId, BalanceOf<T>> {
+				data,
+				provider: who.clone(),
+				deposit,
+			};
+
+            Calls::<T>::insert(sindex, nonce, preimage);
+            Self::deposit_event(Event::<T>::CallSubmitted(supersig_id, nonce, who));
+
+            NonceCall::<T>::insert(sindex, nonce + 1);
+			Ok(())
 		}
 
-		pub fn get_idx_from_id(id: &T::AccountId) -> Option<u128> {
-			PalletId::try_from_sub_account(id).map(|(_, val)| val)
+        // ^
+        // |
+        // |
+        // !!!!!!!!!!!!!!!!!!!!!!!!!!!!
+        //
+        // DEPENDING ON PERFS, ONLY ONE OF THOSE CALL WILL BE KEPT
+        //
+        // !!!!!!!!!!!!!!!!!!!!!!!!!!!!
+        // |
+        // |
+        // v
+
+		#[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
+		pub fn submit_encoded_call(
+			origin: OriginFor<T>,
+			supersig_id: T::AccountId,
+			encoded_call: Vec<u8>,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			let sindex = Self::get_supersig_index_from_id(&supersig_id).ok_or(Error::<T>::SupersigNotFound)?;
+
+			if !Self::is_user_in_supersig(sindex, &who) {
+				return Err(Error::<T>::NotMember.into())
+			}
+			let nonce = Self::nonce_call(sindex);
+			let deposit = <BalanceOf<T>>::from(encoded_call.len() as u32)
+				.saturating_mul(T::PreimageByteDeposit::get());
+
+            T::Currency::reserve(&who, deposit)?;
+
+			let preimage = PreimageCall::<T::AccountId, BalanceOf<T>> {
+				data: encoded_call,
+				provider: who.clone(),
+				deposit,
+			};
+
+            Calls::<T>::insert(sindex, nonce, preimage);
+            Self::deposit_event(Event::<T>::CallSubmitted(supersig_id, nonce, who));
+
+            NonceCall::<T>::insert(sindex, nonce + 1);
+			Ok(())
 		}
 
+		#[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
+		pub fn approve_call(
+			origin: OriginFor<T>,
+			supersig_id: T::AccountId,
+			call_index: CallIndex,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			let sindex = Self::get_supersig_index_from_id(&supersig_id).ok_or(Error::<T>::SupersigNotFound)?;
+
+			if !Self::is_user_in_supersig(sindex, &who) {
+				return Err(Error::<T>::NotMember.into())
+			}
+            if Self::users_votes((sindex, call_index), who.clone()) {
+                return Err(Error::<T>::AlreadyVoted.into())
+            }
+            if Self::calls(sindex, call_index).is_none() {
+                return Err(Error::<T>::CallNotFound.into())
+            }
+
+            UsersVotes::<T>::insert((sindex, call_index), who.clone(), true);
+            Votes::<T>::insert(sindex, call_index, Self::votes(sindex, call_index) + 1);
+
+            Self::deposit_event(Event::<T>::CallVoted(supersig_id, call_index, who));
+
+            let threshold = Self::supersigs(sindex).unwrap().threshold;
```

Comment:
If you unwrap something in production code, always add a comment explaining why this unwrap won't panic in this situation.
It's safe here be cause you check that `sindex < SigNonce`, and we don't have any function to delete a sig.
From there we have two things to do:
- add an external that remove an superSig from the storage, and adding account reference counters to this pallet: https://docs.substrate.io/v3/advanced/account-info/
- remove this `unwrap` and start checking if the sig is Still alive

Reply:
I would say that at multisig creation the creator transfer an amount equal to PreimageByteDeposit (that we rename) * size of the stored data (Supersig + Supersig.members) to the multisig account. This amount is reserved (same way that for PreimageCall).
And when the `delete_multisig` is called, it unreserve those funds and transfer them to an account passed as argument in the call. It also decrement the account reference counters, and remove the superSig from storage. This call being only callable by the superSig itself.

From a high level users can delete their multisig when not used anymore, and are incentivized to do so by this reserved amount